editor/docks/editor_dock_manager.cpp

```diff
@@ -232,6 +232,46 @@ void DockSplitContainer::remove_child_notify(Node *p_child) {
 	_update_visibility();
 }
 
+void DockSplitContainer::set_split_offsets_from_config(Ref<ConfigFile> p_layout, const String &p_section) {
+	PackedInt32Array offsets;
+	int index = 0;
+	for (int i = 0; i < get_child_count(false); i++) {
+		Control *c = Object::cast_to<Control>(get_child(i, false));
+		if (!c || c->is_set_as_top_level()) {
+			continue;
+		}
+		index++;
+		// Only visible children need a split offset, even though all children have a value saved.
+		if (c->is_visible() && p_layout->has_section_key(p_section, "dock_hsplit_" + itos(index))) {
+			int offset = p_layout->get_value(p_section, "dock_hsplit_" + itos(index));
+			offsets.push_back(offset * EDSCALE);
+		}
+	}
+	set_split_offsets(offsets);
+}
+
+PackedInt32Array DockSplitContainer::get_split_offsets_to_save_to_config() const {
+	PackedInt32Array so = get_split_offsets();
+	PackedInt32Array offsets;
+	int index = 0;
+	for (int i = 0; i < get_child_count(false); i++) {
+		Control *c = Object::cast_to<Control>(get_child(i, false));
+		if (!c || c->is_set_as_top_level()) {
+			continue;
+		}
+		if (c->is_visible() && so.size() > index) {
+			offsets.push_back(so[index] / EDSCALE);
+			index++;
+		} else {
+			offsets.push_back(0);
+		}
+		if (offsets.size() >= 4) {
```

Comment:
Is there any way to avoid hard-coding `4`? Why is it even checked? It's not like more children will pass the first condition, no?